src/SourceGenerators/Uno.UI.SourceGenerators/XamlGenerator/XamlCodeGeneration.cs

```diff
@@ -407,7 +407,7 @@ private void TryGenerateUnoResourcesKeyAttribute(ResourceDetailsCollection resou
 			var hasResources = resourceDetailsCollection.HasLocalResources;
 
 			_generatorContext.AddSource(
-				"LocalizationResources",
+				"LocalizationResources.g.cs",
```

Comment:
Isn't this going to produce a `.g.cs.g.cs` file? We're not doing this for any other file, so I'm not sure what the issue is here. 

This file only contains an assembly attribute, and I thought that files produced by generators were automatically excluded from analysis. Is it not the case anymore?

Reply:
@jeromelaban This doesn't produce extra `.g.cs` (as a Roslyn generator, at least). Does uno-based generators append `.g.cs` automatically?

Currently the produced file is `LocalizationResources.cs`. Now it will be `LocalizationResources.g.cs` (nothing is duplicated).

> I thought that files produced by generators were automatically excluded from analysis. Is it not the case anymore?

That was never the case if the generator doesn't produce either an auto-generated comment or make the file name ends with `.g.cs`.

It's common for generators to produce files with `.g.cs`

https://github.com/dotnet/runtime/blob/faab048d14aff308c1fcfb401b89983f883074fb/src/libraries/Microsoft.Extensions.Logging.Abstractions/gen/LoggerMessageGenerator.Roslyn4.0.cs#L57

https://github.com/dotnet/roslyn-sdk/blob/f521744e432429f73e9ae63e156fd39e4f1cba1d/samples/CSharp/SourceGenerators/SourceGeneratorSamples/MustacheGenerator.cs#L33

The relevant code from Roslyn:

https://github.com/dotnet/roslyn/blob/0816b0383e7908effc0b0bdbb0cd97a90bde5a03/src/Compilers/CSharp/Portable/SourceGeneration/CSharpGeneratorDriver.cs#L77

https://github.com/dotnet/roslyn/blob/0816b0383e7908effc0b0bdbb0cd97a90bde5a03/src/Compilers/Core/Portable/SourceGeneration/AdditionalSourcesCollection.cs#L148-L156

This appends only `.cs` (only if it doesn't already exist).

Reply:
`// <auto-generated>` comment would have the exact same effect too. I used `.g.cs` as it is fewer keystrokes, but no objection switching to the comment.

Reply:
I looked into uno generation and it appends `.g.cs` unconditionally (unlike Roslyn).

https://github.com/unoplatform/Uno.SourceGeneration/blob/77cea5f3a108672d2c875a38dfed9408c7d5d1df/src/Uno.SourceGeneration.Engine.Shared/SourceGeneratorEngine.cs#L313

I'll switch to the comment.

Reply:
Thanks for the links!